cmd/typed-errors.go

```diff
@@ -99,7 +99,7 @@ var errInvalidTarget = func(URL string) *probe.Error {
 type targetNotFoundErr error
 
 var errTargetNotFound = func(URL string) *probe.Error {
-	msg := "Target `" + URL + "` not found."
+	msg := "Target `" + URL + "` not found, `rm` expects an object name or ---recursive option to clean up the whole bucket content"
```

Comment:
Are you sure that errTargetNotFound is only used by `mc rm`? Because this will make it unusable with other commands..

Reply:
This is also used by `stat` command, so this is definitely a wrong place. In general, the message here in `errTargetNotFound` is wrong. It should convey that the used URL is either a bucket or a prefix and not Target not found.

Reply:
Best way to solve is making it an abstracted error. Look at `mc mb` for the right errors messages.

The error should arrive from client-s3,fs.go abstraction.